packages/styled/__tests__/styled.js

```diff
@@ -641,6 +641,58 @@ describe('styled', () => {
     const tree = renderer.create(<OneMoreComponent />).toJSON()
     expect(tree).toMatchSnapshot()
   })
+
+  test('withComponent inherits parent shouldForwardProp', () => {
+    const SomeComponent = styled('div', {
+      label: 'cmp-a',
+      shouldForwardProp: prop => prop === 'propA'
+    })``.withComponent('span')
+    const tree = renderer.create(<SomeComponent propA propB />).toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  test('withComponent overrides inherited parent shouldForwardProp', () => {
+    const SomeComponent = styled('div', {
+      label: 'cmp-a',
+      shouldForwardProp: prop => prop === 'propA'
+    })``.withComponent('div', {
+      label: 'cmp-b',
+      shouldForwardProp: prop => prop === 'propB'
+    })
+    const tree = renderer.create(<SomeComponent propA propB />).toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+
+  test('withComponent correctly inherits parent shouldForwardProp through chained withComponent calls', () => {
+    const SomeComponent = styled('div', {
```

Comment:
how is this test different from the previous one? maybe you had in mind a version without `shouldForwardProp` option in the "middle" (in the first `withComponent` call)?